db/types/categories.py

```diff
@@ -0,0 +1,86 @@
+from db.types.base import PostgresType, MathesarCustomType, DatabaseType
+
+STRING_TYPES = frozenset({
+    PostgresType.CHARACTER,
+    PostgresType.CHARACTER_VARYING,
+    PostgresType.TEXT,
+})
+
+STRING_LIKE_TYPES: frozenset[DatabaseType] = frozenset({
+    *STRING_TYPES,
+    PostgresType.CHAR,
+    PostgresType.NAME,
+    MathesarCustomType.URI,
+    MathesarCustomType.EMAIL,
+})
+
+INTEGER_TYPES = frozenset({
+    PostgresType.BIGINT,
+    PostgresType.INTEGER,
+    PostgresType.SMALLINT,
+})
+
+DECIMAL_TYPES = frozenset({
+    PostgresType.DOUBLE_PRECISION,
+    PostgresType.FLOAT,
+    PostgresType.NUMERIC,
+    PostgresType.REAL,
+})
+
+TIME_OF_DAY_TYPES: frozenset[DatabaseType] = frozenset({
+    PostgresType.TIME,
```

Comment:
Hm, not sure. Is it impossible to reflect it?

Reply:
The SQLAlchemy `TIME` type is used for both `TIME WITH TIME ZONE` and `TIME WITHOUT TIME ZONE` db types. We created separate Python classes for those two types. If we _do_ manage to reflect a bare `TIME` Python class rather than one of our custom classes for the type for a column, that would be a bug we should solve.